pkg/cli/cli.go

```diff
@@ -73,6 +74,12 @@ func Init(opts ...Option) *CLI {
 	return cli
 }
 
+// Check if the 'pkl' binary is present in the PATH
+var isPklBinaryPresent = func() bool {
+	_, err := exec.LookPath("pkl")
+	return err == nil
+}
```

Comment:
I would move this as a method of pkl validator

This would help you to validate it's behavior, abdband would prevent race in tests

Reply:
I briefly contemplated this but decided against it for 2 reasons:
- The intended behavior is to skip the files entirely. If we handle this within the pkl validator, we'd have to return a `true` or `false` value, which implies that we've performed the validation and determined whether the file is good or bad. This feels like a breach of contract IMO as it goes against our intention of skipping the files altogether.
- We would lose the ability to log the exact files that were skipped since validate only takes the content of the files, not the file paths

Reply:
I agree with the second point, it would need to rewrite a part of the code

About the idea of perf, I don't mind here. If there is a pkl file, going to the pkl code to see if the file is valid, would require to validate its content. So it would be slow. Of course, when the pkl binary is not present, the code might have to go through the code and finally do not execute the code

Please note you could consider simulating the binary presence, error, absence by calling arbitrary binary name:
- true, always return 0
- false, always return 1
- a random string, the binary will be absent

It would be better than overloading the isPklBinaryPresent by something else

I'm unsure it's a good idea, but why not, it may help

Reply:
I agree with @ccoVeille and feel like the validator should be able to validate if its enabled or not rather than having the check in the CLI. The validator will see the file, realize that it is not set up properly (no PKL) and then skip it and provide that feedback to the report.